src/com/xilinx/rapidwright/timing/delayestimator/DelayEstimatorBase.java

```diff
@@ -0,0 +1,461 @@
+    	}
+    	return 0;
+    }
+
+
+    public short getDelayOf(Node exitNode) {
+	    RoutingNode node = getTermInfo(exitNode);
+	    
+	    // Don't put this in calcTimingGroupDelay because it is called many times to estimate delay.
+	    if (node.ng == T.NodeGroupType.CLE_IN) {
+	    	return inputSitePinDelay.getOrDefault(exitNode.getWireName(), (short) 0);
+	    }
+	    
+	    return calcTimingGroupDelay(node.ng, node.begin(), node.end(), 0d);
+    }
+
+    /**
+     * Represent a routing resource in the delay graph
+     */
+    protected class RoutingNode {
+        // INT_TILE coordinate
+        short x;
+        short y;
+        InterconnectInfo.Direction direction;
+        T.NodeGroupType ng;
+
+
+        RoutingNode() {
+            this.ng = null;
+        }
+
+        void setHorTG(short len) {
+            if (len == 1)
+                this.ng = T.NodeGroupType.valueOf("HORT_SINGLE");
+            else if (len == 2)
+                this.ng = T.NodeGroupType.valueOf("HORT_DOUBLE");
+            else if (len == 4)
+                this.ng = T.NodeGroupType.valueOf("HORT_QUAD");
+            else
+                this.ng = T.NodeGroupType.valueOf("HORT_LONG");
+        }
+
+        void setVerTG(short len) {
+            if (len == 1)
+                this.ng = T.NodeGroupType.valueOf("VERT_SINGLE");
+            else if (len == 2)
+                this.ng = T.NodeGroupType.valueOf("VERT_DOUBLE");
+            else if (len == 4)
+                this.ng = T.NodeGroupType.valueOf("VERT_QUAD");
+            else
+                this.ng = T.NodeGroupType.valueOf("VERT_LONG");
+        }
+        
+        public String toString() {
+            return String.format("x:%d y:%d %s %s", x,y, ng.name(), direction.name());
+        }
+
+        public short begin() {
+            return ng.orientation() == T.Orientation.HORIZONTAL ? x : y;
+        }
+
+        public short end() {
+            short delta = ng.length();
+            short start = ng.orientation() == T.Orientation.HORIZONTAL ? x : y;
+            return  (short) (start + (direction == InterconnectInfo.Direction.U ? delta : -delta));
+        }
+    }
+    
+    private short getLengthFromIntentCode(IntentCode ic) {
+    	switch(ic) {
+    		case NODE_SINGLE:
+    			return 1;
+    		case NODE_DOUBLE:
+    			return 2;
+    		case NODE_VQUAD:
+    		case NODE_HQUAD:
+    			return 4;
+    		case NODE_HLONG:
+    		case NODE_VLONG:
+    			return 12;
+    		default:
+    			return 0;
+    	}
+    }
+    
+    // Currently, the default mode in DelayEstimatorTable is fastMode which use getClosetSrcDstSitePin. 
+    private RoutingNode getTermInfo(Node node) {
+        IntentCode ic = node.getIntentCode();
+        RoutingNode rNode = new RoutingNode();
+
+        rNode.x = (short) node.getTile().getTileXCoordinate();
+        rNode.y = (short) node.getTile().getTileYCoordinate();
+
+
+        String nodeType = node.getWireName();
+        if (nodeType.contains("WW1_E")) {
```

Comment:
This seems quite specific to a use case.  Please add a comment explaining its need or remove.